Clarify JsonProperty unknown field exception

* GINO version: 1.0.0rc2
* Python version: 3.8.1
* asyncpg version: 0.20.1
* aiocontextvars version:
* PostgreSQL version: 11.6

### Description
 `JsonProperty.get_profile` method raises unclear `AttributeError` exception when models attribute name collides with JsonProperty
```
  instance.__profile__[key] = prop.decode(value)
E               AttributeError: 'ColumnAttribute' object has no attribute 'decode'
```



### What I Did

```python

class PropsTest2(db.Model):
    __tablename__ = "props_test2"
    profile = db.Column(JSONB(), nullable=False, server_default="{}")
    bool_profile = db.BooleanProperty()
    bool_attr = db.Column(db.Boolean)
   
await PropsTest2.create(
    profile={"bool_attr": False, "bool_profile": True}, bool_attr=True
)
# bool_attr is defined in the model
# bool_profile is defined as json property
t2 = await PropsTest2.query.gino.first()

assert t2.bool_attr is True
# raises attribute error described above about `bool_attr`
assert t2.bool_profile is True

```

--- a/src/gino/json_support.py
+++ b/src/gino/json_support.py
@@ -60,7 +60,18 @@ class JSONProperty:
                         "`{}` is found in `{}` of instance {}, "
                         "but it is not defined".format(key, self.prop_name, instance)
                     )
-                instance.__profile__[key] = props[key].decode(value)
+
+                prop = props[key]
+                if not isinstance(prop, JSONProperty):
+                    raise UnknownJSONPropertyError(
+                        "`{}` is found in `{}` of instance {}, "
+                        "but column `{}` is not an instance of the "
+                        "`JSONProperty` type.".format(
+                            key, self.prop_name, instance, key
+                        )
+                    )
+                instance.__profile__[key] = prop.decode(value)
+
         return instance.__profile__
 
     def save(self, instance, value=NONE):

--- a/tests/test_json.py
+++ b/tests/test_json.py
@@ -197,6 +197,31 @@ async def test_unknown_properties(bind):
         await PropsTest1.gino.drop()
 
 
+async def test_property_in_profile_and_attribute_collide(bind):
+    from gino.dialects.asyncpg import JSONB
+
+    class PropsTest2(db.Model):
+        __tablename__ = "props_test2"
+        profile = db.Column(JSONB(), nullable=False, server_default="{}")
+        bool_profile = db.BooleanProperty()
+        bool_attr = db.Column(db.Boolean)
+
+    await PropsTest2.gino.create()
+    try:
+        await PropsTest2.create(
+            profile={"bool_attr": False, "bool_profile": True}, bool_attr=True
+        )
+        # bool_attr is defined in the model
+        # bool_profile is defined as json property
+        t2 = await PropsTest2.query.gino.first()
+
+        assert t2.bool_attr is True
+        with pytest.raises(UnknownJSONPropertyError, match=r"bool_attr"):
+            assert t2.bool_profile is True
+    finally:
+        await PropsTest2.gino.drop()
+
+
 async def test_no_profile():
     with pytest.raises(AttributeError, match=r"JSON\[B\] column"):
         # noinspection PyUnusedLocal

